### DISCUSSION TITLE
JWT and App router

### DISCUSSION THREAD

**Discussion by veganmasterclass** [vercel/next.js#69717] [App Router]
I want to migrate von page router to app router. Currently we use for for authorization a JWT (Json Web Token) that we set in every client side request via the middleware.ts file. The JWT comes from a third party server.
So in app router, as far as I understand, we render a lots of files during build. The problem is, during build time we dont know the user. So as a consequence I need to leave the whole application client side rendered.
Do you have another solution for me to integrate JWT with app router?

**Comment by zaharsk** [comment]
Welcome to the club! ^_^

I've been struggling with the storage and update issue for two weeks now. The current status is [here](https://vercel.community/t/help-request-tokens-cookie-refresh-global-issue-in-next-js-14/916). 

If you don't have the challenge of token rotation, it's probably solved through [middleware](https://nextjs-faq.com/server-action-call-inside-server-component-rendering).
